### PR TITLE
Filter by date command

### PR DESCRIPTION
**Pull request by Bogi1001**

### CODE REVIEW

**Comment by pyokagan** [comment]
Ugh, I'm really not free already. Can someone else review? @nightism ? @potatowagon ?

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 1]
```diff
@@ -12,6 +13,8 @@
      * @throws ParseException if the input string could not be parsed. The ranges of the ParseException will
      * contain the substring ranges of the input string which caused the parsing to fail.
      */
+
     T parse(String str) throws ParseException;
 
+
```
Comment (nightism) [inline src/main/java/seedu/address/logic/parser/Parser.java:19]: what is the extra space for?

File: src/main/java/seedu/address/logic/parser/FilterByDateCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import java.time.LocalTime;
+import java.util.Optional;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.model.filter.FilterByDate;
+import seedu.address.model.filter.TaskPredicate;
+
+public class FilterByDateCommandParser implements Parser<Command>{
+
```
Comment (nightism) [inline src/main/java/seedu/address/logic/parser/FilterByDateCommandParser.java:14]: I only saw a Command Parser, but where is the Command?

Comment (nightism) [reply src/main/java/seedu/address/logic/parser/FilterByDateCommandParser.java:14]: That's not how our `Command` works, the `CommandParser.parse()` parse a command line and then return a new `Command` object.

File: src/main/java/seedu/address/model/filter/FilterByDate.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package seedu.address.model.filter;
+
+import java.time.LocalDateTime;
+import java.util.stream.Stream;
+
+import seedu.address.model.task.DeadlineTask;
+import seedu.address.model.task.EventTask;
+import seedu.address.model.task.FloatingTask;
+
+public class FilterByDate implements TaskPredicate{
```
Comment (nightism) [inline src/main/java/seedu/address/model/filter/FilterByDate.java:10]: This class name is quite confusing. And this class should have some heading descriptions.

File: src/main/java/seedu/address/storage/config/JsonConfigStorage.java [hunk 4]
```diff
@@ -82,4 +82,4 @@ public void saveConfig(ReadOnlyConfig config, String filePath) throws IOExceptio
         objectMapper.writeValue(file, config);
     }
 
-}
+}
```
Comment (nightism) [inline src/main/java/seedu/address/storage/config/JsonConfigStorage.java:85]: Please do stylecheck, the .java file should end with an empty line.

**Review by nightism — CHANGES_REQUESTED** [review]
some comments added

**Comment by Bogi1001** [comment]
Okay. Thanks for the feedback, I shall implement the changes ... Also, do u have any feedback about the AddDeadlineCommandTest.java that I added?

**Pull request closed without merging**